agent/src/runtime_connectors/podman_cli.rs

```diff
@@ -616,17 +749,39 @@ mod tests {
                     "label=name=test_agent",
                     "--format=json",
                 ])
-                .exec_returns(Ok(generate_container_info_json(
-                    PodmanContainerState::Running,
-                    0,
-                    "workload_name".into(),
-                ))),
+                .exec_returns(Ok([TestPodmanContainerInfo {
+                    labels: &[("name", "workload_name")],
+                    ..Default::default()
+                }]
+                .to_json())),
         );
 
         let res = PodmanCli::list_workload_names_by_label("name", "test_agent").await;
         assert_eq!(res, Ok(vec!["workload_name".into()]));
     }
 
+    #[tokio::test]
+    async fn utest_list_workload_names_no_name_label() {
```

Comment:
Maybe rename the test to:
```suggestion
    async fn utest_list_workload_names_not_found_success() {
```